File: spec/javascript/test_expr_mod.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ExprMod', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.intU, 1262698832);
  assert.equal(r.intS, -52947);

  assert.equal(r.modPosConst, 9);
  assert.equal(r.modNegConst, 4);
  assert.equal(r.modPosSeq, 5);
  assert.equal(r.modNegSeq, 2);
});

File: spec/python/test_expr_mod.py
import unittest

from expr_mod import ExprMod

class TestExprMod(unittest.TestCase):
    def test_expr_mod(self):
        r = ExprMod.from_file("src/fixed_struct.bin")

        self.assertEquals(r.int_u, 1262698832)
        self.assertEquals(r.int_s, -52947)

        self.assertEquals(r.mod_pos_const, 9)
        self.assertEquals(r.mod_neg_const, 4)
        self.assertEquals(r.mod_pos_seq, 5)
        self.assertEquals(r.mod_neg_seq, 2)
